ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/screen/request/view/RequestViewTest.kt

```diff
@@ -81,49 +73,91 @@ class RequestViewTest {
         assertEquals(null, testSetup.getLastCreateZecRequest())
 
-        composeTestRule.onNodeWithText(getStringResource(R.string.request_amount)).also {
-            val separators = MonetarySeparators.current()
+        composeTestRule.setValidAmount()
 
-            it.performTextInput("{${separators.decimal}}123")
-        }
-
-        composeTestRule.onNodeWithText(getStringResource(R.string.request_message)).also {
-            it.performTextInput(ZecRequestFixture.MESSAGE.value)
-        }
+        composeTestRule.setValidMessage()
 
-        composeTestRule.onNodeWithText(getStringResource(R.string.request_create)).also {
-            it.performClick()
-        }
+        composeTestRule.clickCreateAndSend()
 
         assertEquals(1, testSetup.getOnCreateCount())
 
         testSetup.getLastCreateZecRequest().also {
             assertNotNull(it)
             assertEquals(WalletAddressFixture.unified(), it.address)
-            assertEquals(Zatoshi(12300000), it.amount)
+            assertEquals(Zatoshi(12345600000), it.amount)
             assertEquals(ZecRequestFixture.MESSAGE.value, it.message.value)
         }
     }
 
     @Test
     @MediumTest
-    @Ignore("https://github.com/zcash/secant-android-wallet/issues/218")
-    fun create_request_illegal_input() {
+    fun check_regex_functionality_valid_inputs() {
         val testSetup = TestSetup(composeTestRule)
+        val separators = MonetarySeparators.current()
 
         assertEquals(0, testSetup.getOnCreateCount())
         assertEquals(null, testSetup.getLastCreateZecRequest())
 
-        composeTestRule.onNodeWithText(getStringResource(R.string.request_amount)).also {
-            val separators = MonetarySeparators.current()
+        composeTestRule.setAmount("123")
+        composeTestRule.clickCreateAndSend()
+        assertEquals(1, testSetup.getOnCreateCount())
 
-            it.performTextInput("{${separators.decimal}}1{${separators.decimal}}2{${separators.decimal}}3{${separators.decimal}}4")
-        }
+        // e.g. 123,
+        composeTestRule.setAmount("123${separators.grouping}")
+        composeTestRule.clickCreateAndSend()
+        assertEquals(2, testSetup.getOnCreateCount())
+
+        // e.g. 123.
+        composeTestRule.setAmount("123${separators.decimal}")
+        composeTestRule.clickCreateAndSend()
+        assertEquals(3, testSetup.getOnCreateCount())
+
+        // e.g. 123,456.
+        composeTestRule.setAmount("123${separators.grouping}456${separators.decimal}")
+        composeTestRule.clickCreateAndSend()
+        assertEquals(4, testSetup.getOnCreateCount())
+
+        // e.g. 123,456.789
+        composeTestRule.setAmount("123${separators.grouping}456${separators.decimal}789")
+        composeTestRule.clickCreateAndSend()
+        assertEquals(5, testSetup.getOnCreateCount())
+    }
 
-        composeTestRule.onNodeWithText(getStringResource(R.string.request_create)).also {
-            it.performClick()
-        }
+    @Test
```

Comment:
Ok, understand. I check it and we'll see.

Reply:
@ccjernigan Now I've enhanced the regex the way it allows us to input decimals `.123` and then the subsequent formatting with `Zatoshi.Companion.fromZecString` results like this `0.12300000`. So that seems correct to me.